src/txn.ts

```diff
@@ -52,18 +52,25 @@ export class Txn {
     public async queryWithVars(
         q: string,
         vars?: { [k: string]: any }, // tslint:disable-line no-any
-        options: { debug?: boolean } = {},
+        options: { debug?: boolean, readOnly?: boolean, bestEffort?: boolean } = {},
     ): Promise<Response> {
         if (this.finished) {
             this.dc.debug(`Query request (ERR_FINISHED):\nquery = ${q}\nvars = ${vars}`);
             throw ERR_FINISHED;
         }
 
+        if (options.bestEffort && !options.readOnly) {
+            this.dc.debug('Best effort only works with read-only queries.');
+            throw ERR_FINISHED;
```

Comment:
doesn't sound like the right ERR object to throw

Reply:
Good point

Reply:
Done